packages/components/aside/src/aside.js

```diff
@@ -0,0 +1,67 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import css from './aside.css';
+import '@tradeshift/elements.button';
+import '@tradeshift/elements.cover';
+
+customElementDefineHelper(
+	'ts-aside',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				title: { type: String, attribute: 'data-title' },
+				visible: { type: Boolean, attribute: 'data-visible', reflect: true },
+				hasFoot: { type: Boolean }
+			};
+		}
+
+		constructor() {
+			super();
+			this.title = '';
+			this.visible = false;
+			this.hasFoot = false;
+		}
+
+		close(e) {
+			this.visible = false;
+		}
+
+		footerSlot(e) {
+			this.hasFoot = true;
+		}
+
+		get direction() {
+			return this.bodyHasRTL ? 'rtl' : 'ltr';
+		}
+
+		get slide() {
+			const pre = 'ts-slide';
+			const mid = this.visible ? 'in' : 'out';
+			const dir = this.bodyHasRTL ? 'left' : 'right';
```

Comment:
You think it's a better idea to just have the direction check on the body, instead of the components themselves?

Reply:
Good idea.